firebase-database/src/main/java/com/google/firebase/database/ServerValue.java

```diff
@@ -30,11 +32,49 @@
    * milliseconds) by the Firebase Database servers.
    */
   @NonNull
-  public static final Map<String, String> TIMESTAMP = createServerValuePlaceholder("timestamp");
+  public static final Map<String, String> TIMESTAMP = createScalarServerValuePlaceholder("timestamp");
 
-  private static Map<String, String> createServerValuePlaceholder(String key) {
+  /**
+   * Adds the given delta to the ref's current value.
+   *
+   * This must be an long or a double value. If the ref is not an integer or double, or if the ref
+   * does not yet exist, the transformation will set the ref to the given value. If either of the
+   * given value or the current field value are doubles, both values will be interpreted as doubles.
+   * Double arithmetic and representation of double values follow IEEE 754 semantics. If there is
+   * positive/negative integer overflow, the field is resolved to a double.
+   * @param delta the amount to modify the ref's current value atomically.
+   * @return a placeholder value for modifying a ref atomically server-side.
+   */
+  static final Map<String, Map<String, Object>> increment(long delta) {
```

Comment:
Is this return type used anywhere? The type seems very strange.

Would it be possible to return an Object here? You could create an private subclass that you can use internally to allow for `instanceof` checks.

Reply:
This was intended for consistency with TIMESTAMP which is also a transparent return type. Can change.

Reply:
I would prefer is this just returned `Object`. `Map<String, String>` is still a little more meaningful that `Map< String, Map<String, Object>>`

Reply:
Done. The helper type is still a Map, but the public methods return Object.